ethereum/contracts/zksync/libraries/TransactionValidator.sol

```diff
@@ -15,21 +15,19 @@ library TransactionValidator {
     /// @param _transaction The transaction to validate
     /// @param _encoded The abi encoded bytes of the transaction
     /// @param _priorityTxMaxGasLimit The max gas limit, generally provided from Storage.sol
+    /// @param _priorityTxMaxPubdata The maximal amount of pubdata that a single L1->L2 transaction can emit
     function validateL1ToL2Transaction(
         IMailbox.L2CanonicalTransaction memory _transaction,
         bytes memory _encoded,
-        uint256 _priorityTxMaxGasLimit
+        uint256 _priorityTxMaxGasLimit,
+        uint256 _priorityTxMaxPubdata
     ) internal pure {
-        uint256 l2GasForTxBody = getTransactionBodyGasLimit(
-            _transaction.gasLimit,
-            _transaction.gasPerPubdataByteLimit,
-            _encoded.length
-        );
+        uint256 l2GasForTxBody = getTransactionBodyGasLimit(_transaction.gasLimit, _encoded.length);
 
         // Ensuring that the transaction is provable
         require(l2GasForTxBody <= _priorityTxMaxGasLimit, "ui");
         // Ensuring that the transaction cannot output more pubdata than is processable
-        require(l2GasForTxBody / _transaction.gasPerPubdataByteLimit <= PRIORITY_TX_MAX_PUBDATA, "uk");
+        require(l2GasForTxBody / _transaction.gasPerPubdataByteLimit <= _priorityTxMaxPubdata, "uk");
```

Comment:
Do you think it makes to add a comment that we intentionally don't check that the tx encoding length is less than the maximal allowed? (As I understand it's just impossible to exceed this limit, due to gas limitations on l1)